cds/modules/ldap/views.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016, 2017 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""CDS Stats."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, current_app, jsonify, make_response, request, \
+    abort
+
+from cds.modules.ldap.client import LdapClient
+from cds.modules.ldap.decorators import needs_authentication
+from cds.modules.ldap.serializers import serialize_ldap_users
+
+
+blueprint = Blueprint(
+    'cds_ldap',
+    __name__,
+    url_prefix='/ldap/',
+)
+
+
+@blueprint.route('users/', methods=["GET"])
+@needs_authentication
+def get_users():
+    query = request.args.get('query', '')
+    if query:
+        ldap_client = LdapClient()
```

Comment:
I would keep in memory for a few minutes the object `LdapClient`, WDYT?

Reply:
I explored the possibility, and there are two solutions (both costly in terms of time):
1. I serialize the object to be saved in redis (unlikely to be doable)
2. I try to attach second caching backend (i.e. memcache) because we are using only redis now.

or any other solution I didn't consider